app/controllers/campaigns_controller.rb

```diff
@@ -45,6 +48,11 @@ def destroy
 
   private
 
+  def load_commentable
+    resource, id = request.path.split('/')[2, 3]
+    @commentable = resource.singularize.classify.constantize.find(id)
```

Comment:
since we are in the campaigns controller it is obvious that commentable is Campaign, is not it? Then we do not need to parse the URL

Reply:
You can rename it to load comments and have something like Campaign.find_by(:id)&.comments